src/pages/build-apps/guides/authentication.md

```diff
@@ -0,0 +1,250 @@
+authenticator encrypts secret data such as the app private key using this
+public key and sends it back to the app when the user signs in to the app. The
+transit private key signs the app authentication request.
+
+### Identity address private key
+
+The identity address private key is derived from the user's keychain phrase and
+is the private key of the Stacks username that the user chooses to use to sign in
+to the app. It is a secret owned by the user and never leaves the user's
+instance of the authenticator.
+
+This private key signs the authentication response token for an app to indicate that the user approves sign in to that app.
+
+### App private key
+
+The app private key is an app-specific private key that is generated from the
+user's identity address private key using the `domain_name` as input. It is
+deterministic in that for a given Stacks username and `domain_name`, the same
+private key is generated each time.
+
+The app private key is securely shared with the app on each authentication, encrypted by the authenticator with the transit public key.
+
+## authRequest Payload Schema
+
+```jsx
+const requestPayload = {
+  jti, // UUID
+  iat, // JWT creation time in seconds
+  exp, // JWT expiration time in seconds
+  iss, // legacy decentralized identifier generated from transit key
+  public_keys, // single entry array with public key of transit key
+  domain_name, // app origin
+  manifest_uri, // url to manifest file - must be hosted on app origin
+  redirect_uri, // url to which the authenticator redirects user on auth approval - must be hosted on app origin
+  version, // version tuple
+  do_not_include_profile, // a boolean flag asking authenticator to send profile url instead of profile object
+  supports_hub_url, // a boolean flag indicating gaia hub support
+  scopes, // an array of string values indicating scopes requested by the app
+};
+```
+
+## authResponse Payload Schema
+
+```jsx
+const responsePayload = {
+  jti, // UUID
+  iat, // JWT creation time in seconds
+  exp, // JWT expiration time in seconds
+  iss, // legacy decentralized identifier (string prefix + identity address) - this uniquely identifies the user
+  private_key, // encrypted private key payload
+  public_keys, // single entry array with public key
+  profile, // profile object or null if passed by profile_url
+  username, // Stacks username (if any)
+  core_token, // encrypted core token payload
+  email, // email if email scope is requested & email available
+  profile_url, // url to signed profile token
+  hubUrl, // url pointing to user's gaia hub
+  version, // version tuple
+};
+```
+
+## Decode authRequest or authResponse
+
+To decode a token and see what data it holds:
+
+1. Copy the `authRequest` or `authResponse` string from the URL during authentication.
+2. Navigate to [jwt.io](https://jwt.io/).
+3. Paste the full token there.
+
+   The output should look similar to below:
+
+   ```json
+   {
```

Comment:
The encrypted private key is also returned here - called `private_key`.

Reply:
Added:

![image](https://user-images.githubusercontent.com/28991/104126120-23a3d180-535b-11eb-8b42-4781f268e52b.png)